test/unittests/execute_numeric_test.cpp

```diff
@@ -519,15 +519,15 @@ TEST(execute_numeric, i64_extend_i32_u_2)
 TEST(execute_numeric, i64_clz)
 {
     const auto i64_clz = create_unary_operation_executor(Instr::i64_clz);
-    EXPECT_THAT(i64_clz(0_u64), Result(64));
-    EXPECT_THAT(i64_clz(0x7f_u64), Result(64 - 7));
+    for (const auto& [input, expected] : countl_zero64_test_cases)
+        EXPECT_THAT(i64_clz(input), Result(expected)) << input;
 }
 
 TEST(execute_numeric, i64_ctz)
 {
     const auto i64_ctz = create_unary_operation_executor(Instr::i64_ctz);
-    EXPECT_THAT(i64_ctz(0_u64), Result(64));
-    EXPECT_THAT(i64_ctz(0x80_u64), Result(7));
+    for (const auto& [input, expected] : countr_zero64_test_cases)
+        EXPECT_THAT(i64_ctz(input), Result(expected)) << input;
```

Comment:
Perhaps this is the correct one, as `i64.ctz` returns `u32`?

```suggestion
        EXPECT_THAT(i64_ctz(input), Result(static_cast<uint32_t>(expected))) << input;
```

Reply:
Or we change all the test cases to always return `uint32_t`, as even though the c++ bit implementations are defined as `T countl_zero(T)`, they never return anything big?

Reply:
Missed that it is actually `int countl_zero(T)`, only wasm does otherwise. So changed all to use `int` properly.